demo/react/src/VoiceWidget.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useCallback, useRef } from "react";
+import { useLeopard } from "@picovoice/leopard-react";
+
+import leopardModel from "./lib/leopardModel";
+
+export default function VoiceWidget() {
+  const accessKeyRef = useRef<string>("");
+
+  const {
+    init,
+    result,
+    isLoaded,
+    processFile,
+    startRecording,
+    stopRecording,
+    isRecording,
+    recordingElapsedSec,
+    release,
+    error,
+  } = useLeopard();
+
+  const initEngine = useCallback(async () => {
+    if (accessKeyRef.current.length === 0) {
+      return;
+    }
+
+    await init(
+      accessKeyRef.current,
+      leopardModel,
+      {
+        enableAutomaticPunctuation: true,
+      },
+      {
+        transfer: true,
+      }
+    );
+  }, [init]);
+
+  const toggleRecord = async () => {
+    if (isRecording) {
+      await stopRecording();
+    } else {
+      await startRecording();
+    }
+  };
+
+  return (
+    <div className="voice-widget">
+      <h2>VoiceWidget</h2>
+      <h3>
+        <label>
+          AccessKey obtained from{" "}
+          <a href="https://console.picovoice.ai/">Picovoice Console</a>:{" "}
+          <input
+            type="text"
+            name="accessKey"
+            onChange={(e) => {
+              accessKeyRef.current = e.target.value;
+            }}
+          />
+          <button
+            className="init-button"
+            onClick={initEngine}
```

Comment:
since initialization and transcription for Leopard take a significant amount of time, you are going to want to disable buttons that you don't want the user to press while it's busy. For instance, I can click initialize and start recording one after another and cause an error.